chore(docs): Document spacing between components (#2455)
Amsterdam/design-system: storybook/src/docs/designer-guide/space-between.docs.mdx

{/* @license CC0-1.0 */}

import { Meta } from "@storybook/addon-docs/blocks";
import { SpaceBetweenFinder } from "../../_components/SpaceBetweenFinder/SpaceBetweenFinder";

<Meta title="Docs/Designer Guide/Space between" />

# Space between

Components handle their own internal whitespace, but the space between them is yours to apply.

> See also: [‘Spacing Research’ pages](http://localhost:6006/?path=/docs/pages-public-spacing-research--docs&args=showMarkers:!false)

Author: Fixed

- Misschien in de tabel P en P large samenvoegen, met een notitie dat onder een H1 altijd een P large moet ← Nee, onduidelijk
- Kunnen we een groep ‘Grote componenten maken’ waar bijv. Image in zit? ← Zit in plan
- En dan Sectie achtige dingen zoals Progress List ← Zit in plan
- En Figure, met Caption? ← Zit in plan

### To do

- Table met Caption aanraden, dan Heading gebruiken voor spacing ← Done, nog checken
- Table zonder Caption is dan wat er in de tabel komt ← Done, nog checken
- Overwegen om Heading 2 --> Heading een l te maken voor het aflopende trappetje ← Zit in plan, check UX
- Onderscheid maken tussen Image (m?) en Hero Image (xl) ← Nagaan of dat goed in Grid Cell docs staat

### Later

- Componenten Content Header en Content Body in Figma. Ook iets voor Code?
- Breadcrumb container binnenkort mee aan de slag? (Drupal)
- Token of Variable voor spacing 0 lijkt nog niet nodig?
- Figma ticket voor klaarzetten Content patroontjes

## Templates te maken

- Contentpagina met h1 en h2 achter elkaar (zonder foto en intro) dus eigenlijk Czaar Peter

> -

Author: Sure.

{/* @license CC0-1.0 */}

import { Meta } from "@storybook/addon-docs/blocks";
import { SpaceBetweenFinder } from "../../_components/SpaceBetweenFinder/SpaceBetweenFinder";

<Meta title="Docs/Designer Guide/Space between" />

# Space between

Components handle their own internal whitespace, but the space between them is yours to apply.

See also: [‘Spacing Research’ pages](http://localhost:6006/?path=/docs/pages-public-spacing-research--docs&args=showMarkers:!false)

## Design

- Spacing is responsive and differs per [mode](/docs/docs-developer-guide-modes--docs).
  That’s why we use t-shirt sizes, not pixel values.
- In general, space between two equal or comparable components is medium.
- Space between containers is large.
  Space between grid columns is x-large.

## Component families

We sort components into five families.
Most spacing rules apply to a whole family, not to individual components.

**1. Heading 1 + Lead Paragraph** open a page.
A Heading 1 is always followed by a Paragraph (large).

**2. Heading 2–4** structure the content.
The space _above_ a heading depends only on its level.

The remaining three families are defined by their members:

> | 3. Body content  | 4. Block content | 5. Own Grid Cell |

Reviewer: ```suggestion
| 3. Body text | 4. Body blocks | 5. Own Grid Cell |
```

“Body blocks” could remain “Block content”. Not sure about that yet.

| :--------------- | :--------------- | :--------------- |
| Paragraph        | Blockquote       | Alert            |
| Description List | Image            | Breadcrumb       |
| Link List        | Table            | Breakout         |
| Ordered List     |                  | Image Slider     |
| Unordered List   |                  | Spotlight        |
| Standalone Link  |                  |                  |
| Button           |                  |                  |
| CTA Link         |                  |                  |

> - **Body content** carries the running content; items sit close together.

Reviewer: ```suggestion
- **Body text** carries the running content; items sit close together.
```

> - **Block content** interrupts the flow with a stand-out block; items need more breathing room.

Reviewer: ```suggestion
- **Body blocks** interrupts the flow with a stand-out block; items need more breathing room.
```

### Space below other components

|                           | Button / CTA Link | Image | … List | Standalone Link | Table |
| :------------------------ | :---------------: | :---: | :----: | :-------------: | :---: |
| **Heading 2**             |       `xl`        | `xl`  |  `xl`  |      `xl`       | `xl`  |
| **Heading 3**             |        `l`        |  `l`  |  `l`   |       `l`       |  `l`  |
| **Heading 4**             |        `l`        |  `l`  |  `m`   |       `l`       |  `l`  |
| **Blockquote**            |        `l`        |  `l`  |  `l`   |       `l`       |  `l`  |
| **Button** / **CTA Link** |        `l`        |  `l`  |  `m`   |       `m`       |  `l`  |
| **Image**                 |        `l`        |  `l`  |  `l`   |       `l`       |  `l`  |
| **Standalone Link**       |        `l`        |  `l`  |  `m`   |       `m`       |  `l`  |
| **… List**                |        `l`        |  `l`  |  `m`   |       `m`       |  `l`  |
| **Paragraph**             |        `l`        |  `l`  |  `m`   |       `m`       |  `l`  |
| **Table**                 |        `l`        |  `l`  |  `l`   |       `l`       |  `l`  |

## Code example

> {/* prettier-ignore */}

Reviewer 2: This is weird, the prettier config does not include markdown files.

Author: It does somehow; adds semicolons that get a red highlight in Storybook.

```tsx

> <Grid.Cell>

Reviewer 2: Not recommending the prose util?

Author: The final version shows both: http://localhost:6006/?path=/docs/docs-designer-guide-vertical-space--docs#code-examples

</Grid.Cell>
```

## Notes

### Done

- Space `2xl` moet er eigenlijk uit, die willen we alleen voor Grid horizontaal gebruiken, dus geen token ook. (Of toch Grid verticaal?) ← Done
- Misschien in de tabel P en P large samenvoegen, met een notitie dat onder een H1 altijd een P large moet ← Nee, onduidelijk
- Kunnen we een groep ‘Grote componenten maken’ waar bijv. Image in zit? ← Zit in plan
- En dan Sectie achtige dingen zoals Progress List ← Zit in plan
- En Figure, met Caption? ← Zit in plan

### To do

- Table met Caption aanraden, dan Heading gebruiken voor spacing ← Done, nog checken
- Table zonder Caption is dan wat er in de tabel komt ← Done, nog checken
- Overwegen om Heading 2 --> Heading een l te maken voor het aflopende trappetje ← Zit in plan, check UX
- Onderscheid maken tussen Image (m?) en Hero Image (xl) ← Nagaan of dat goed in Grid Cell docs staat

### Later

- Componenten Content Header en Content Body in Figma. Ook iets voor Code?
- Breadcrumb container binnenkort mee aan de slag? (Drupal)
- Token of Variable voor spacing 0 lijkt nog niet nodig?
- Figma ticket voor klaarzetten Content patroontjes

## Templates te maken

> - Contentpagina met h1 en h2 achter elkaar (zonder foto en intro) dus eigenlijk Czaar Peter

Reviewer 2: Is this unfinished?

Author: Yes, they were work notes.
